### PR TITLE
Add Discover Table Tests

### PR DESCRIPTION
**Pull request by LDrago27**
### Description

This adds functional tests for the issues described in https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6280.

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5713

Issues Still Pending (Sub Issues within the Test Suite are mentioned within brackets)

1. **Function Test Suite 2: Copy data without extra data:** No easy way to test copying data to clipboard in cypress
2. **Function Test Suite 5: Left Navigation Display (Should display Unknown type field properly in left navigation):** Niche scenario
3. **Function Test Suite 6: Saved Search Embeddable in Dashboard (When load a saved search in dashboard, sort should be loaded properly Issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5933)** : Issue still open hence couldn't be tested
4. **Function Test Suite 10: Saved query (From index manage page (saved object), click a saved query object created from Discover/Dashboard/Vis can load Discover with proper query and filter Issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5942)** : Issue Still open hence can't be tested

The test suites are divided across two files discover_table and discover_advanced_settings. All the functional tests related to the Advanced Settings are grouped in discover_advanced_setting while the rest are part of discover_table settings.

**Tests when run without security**

https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/38322563/b7d4ead2-99e2-45e0-a7eb-79d545a01e8a

https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/38322563/8a636136-6473-4c46-ada7-0ef06a67ad51

**Tests when run with security**

https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/38322563/35cae8a4-70b8-4d97-8154-252f44c2e9ce

https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/38322563/1eded379-9e6c-4cb9-9074-9cd02493e4a1

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    // Go to the Discover page
+    miscUtils.visitPage(
+      `app/data-explorer/discover#/?_g=(filters:!(),time:(from:'2015-09-19T13:31:44.000Z',to:'2015-09-24T01:31:44.000Z'))`
+    );
+    cy.waitForSearch();
+    cy.waitForLoader();
+  });
+
+  after(() => {});
```
Comment (Hailong-am) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js:65]: can be removed?

**Review by Hailong-am — APPROVED** [review]

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js (hunk 1 above)
Comment (LDrago27) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js]: Removed it in the latest version of the PR.

**Review by Hailong-am — APPROVED** [review]

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+const indexSet = [
+  'logstash-2015.09.22',
+  'logstash-2015.09.21',
+  'logstash-2015.09.20',
+];
+
+// Setting up the page
+describe('discover_table', () => {
+  beforeEach(() => {
+    if (Cypress.env('SECURITY_ENABLED')) {
```
[the remark this answers is not in the document]
Comment (LDrago27) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js]: I have updated both the tests to remove the check specific to security.

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    });
+
+    // Go to the Discover page
+    miscUtils.visitPage(
+      `app/data-explorer/discover#/?_g=(filters:!(),time:(from:'2015-09-19T13:31:44.000Z',to:'2015-09-24T01:31:44.000Z'))`
+    );
+    cy.waitForSearch();
+  });
+
+  describe('no line wrapping in legacy table', () => {
```
[the remark this answers is not in the document]
Comment (LDrago27) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js]: Updated the description.

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    it('checks that there is no line wrapping by default in the legacy table', function () {
```
[the remark this answers is not in the document]
Comment (LDrago27) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js]: Updated the description.

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    });
+
+    it('checks if the table respects maxHeight setting of truncation', function () {
+      cy.get('.truncate-by-height')
+        .first()
+        .should('have.css', 'max-height', '130px');
+    });
+  });
+
+  describe('maxHeight With No truncation', () => {
```
[the remark this answers is not in the document]
Comment (LDrago27) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js]: Consolidated both the tests into one.

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            .contains('Display')
+            .click()
+            .then(() => {
+              cy.get('[type="number"]').type('{selectall}').type('2'); // Update the line count value
+            });
+
+          cy.get('[data-test-subj="dataGridRowCell"]')
+            .first()
+            .invoke('height')
+            .should('be.gt', res); // Check if the new height is greater than previus height
```
[the remark this answers is not in the document]
Comment (LDrago27) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/discover_table.spec.js:107]: All the tests related to the table will be housed within this file. This PR only covers the tests corresponding to https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6280. There will be follow up PR addressing other table related issues.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by SuZhou-Joe — CHANGES_REQUESTED** [review]
Seems the new added test cases failed to pass, please take a look here.

**Comment by LDrago27** [comment]
@SuZhou-Joe I have updated the tests to fix the failures. The newly added tests are now passing for both with and without security enabled.

**Comment by SuZhou-Joe** [comment]
@LDrago27 Thanks for addressing all the failed cases.

**Pull request merged**